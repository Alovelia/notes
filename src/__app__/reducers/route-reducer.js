import { constants, actions } from 'ducks-helpers';
import handleActions from 'redux-actions/lib/handleActions';
import { LOCATION_CHANGE } from 'react-router-redux';
import _ from 'common/helpers';

export const TYPE = constants('@@router', [
  'UPDATE_PARAMS',
  //†type
]);
export const ACTION = actions(TYPE);

// Initial routing state
export const initialState = _.fromJS({
  locationBeforeTransitions: null,
  params: {}
});

/**
 * Merge route into the global application state
 */
export default handleActions({
  [LOCATION_CHANGE]: updateLocation,
  [TYPE.UPDATE_PARAMS]: updateParams,
  //†handler
}, initialState);

/*
 * routeReducer
 * The reducer merges route location changes into our immutable state.
 * The change is necessitated by moving to react-router-redux@4
 */
export function updateLocation(state, { payload }) {
  return state.merge({
    locationBeforeTransitions: payload,
  });
}

export function updateParams(state, { payload }) {
  return state.set('params', _.fromJS(payload));
}
//†reducer
