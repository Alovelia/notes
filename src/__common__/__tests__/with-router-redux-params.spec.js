import { shallow } from 'enzyme';
import React from 'react';
import noop from 'lodash/noop';
import { ACTION } from 'app/reducers/route-reducer';
import withRourReduxParams from '../with-router-redux-params';

describe('<WithRouterRedux />', () => {
  let wrapper;
  let stub;
  beforeAll(() => {
    const mockComponent = (props) => <div>MOCK</div>;
    wrapper = shallow(withRourReduxParams(mockComponent));
    // stub = sinon.stub(ACTION, 'get').returns({ type: 'mock' });
  });
  afterAll(() => {
    // stub.restore();
  });
  it('when initializing, the expected output is rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
