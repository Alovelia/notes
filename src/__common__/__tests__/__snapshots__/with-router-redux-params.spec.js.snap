// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithRouterRedux /> when initializing, the expected output is rendered 1`] = `undefined`;
